From cb21af7f69fc44572c9f9a326c84275ae2be9c63 Mon Sep 17 00:00:00 2001
From: Rob Murray <rob.murray@docker.com>
Date: Wed, 13 Mar 2024 14:54:25 +0000
Subject: [PATCH 1/3] Fix tests that look at 'Aliases'

Inspect output for 'NetworkSettings.Networks.<network>.Aliases'
includes the container's short-id (although it will be removed
in API v1.45, in moby 26.0).

Signed-off-by: Rob Murray <rob.murray@docker.com>
---
 tests/integration/models_containers_test.py | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/tests/integration/models_containers_test.py b/tests/integration/models_containers_test.py
index 219b9a4cb1..87ba89b1de 100644
--- a/tests/integration/models_containers_test.py
+++ b/tests/integration/models_containers_test.py
@@ -110,12 +110,12 @@ def test_run_with_networking_config(self):
         client.networks.create(net_name)
         self.tmp_networks.append(net_name)
 
-        test_aliases = ['hello']
+        test_alias = 'hello'
         test_driver_opt = {'key1': 'a'}
 
         networking_config = {
             net_name: client.api.create_endpoint_config(
-                aliases=test_aliases,
+                aliases=[test_alias],
                 driver_opt=test_driver_opt
             )
         }
@@ -132,8 +132,10 @@ def test_run_with_networking_config(self):
         assert 'NetworkSettings' in attrs
         assert 'Networks' in attrs['NetworkSettings']
         assert list(attrs['NetworkSettings']['Networks'].keys()) == [net_name]
-        assert attrs['NetworkSettings']['Networks'][net_name]['Aliases'] == \
-               test_aliases
+        # Expect Aliases to list 'test_alias' and the container's short-id.
+        # In API version 1.45, the short-id will be removed.
+        assert attrs['NetworkSettings']['Networks'][net_name]['Aliases'] \
+               == [test_alias, attrs['Id'][:12]]
         assert attrs['NetworkSettings']['Networks'][net_name]['DriverOpts'] \
                == test_driver_opt
 
@@ -190,7 +192,9 @@ def test_run_with_networking_config_only_undeclared_network(self):
         assert 'NetworkSettings' in attrs
         assert 'Networks' in attrs['NetworkSettings']
         assert list(attrs['NetworkSettings']['Networks'].keys()) == [net_name]
-        assert attrs['NetworkSettings']['Networks'][net_name]['Aliases'] is None
+        # Aliases should include the container's short-id (but it will be removed
+        # in API v1.45).
+        assert attrs['NetworkSettings']['Networks'][net_name]['Aliases'] == [attrs["Id"][:12]]
         assert (attrs['NetworkSettings']['Networks'][net_name]['DriverOpts']
                 is None)
 

From e91b280074784026135573ab1726a565c090cd82 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pawe=C5=82=20Gronowski?= <pawel.gronowski@docker.com>
Date: Thu, 7 Mar 2024 13:12:32 +0100
Subject: [PATCH 2/3] Bump default API version to 1.44 (Moby 25.0)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>
---
 Makefile                  | 4 ++--
 docker/constants.py       | 2 +-
 tests/Dockerfile-ssh-dind | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 00ebca05ce..25a83205b6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
-TEST_API_VERSION ?= 1.43
-TEST_ENGINE_VERSION ?= 24.0
+TEST_API_VERSION ?= 1.44
+TEST_ENGINE_VERSION ?= 25.0
 
 ifeq ($(OS),Windows_NT)
     PLATFORM := Windows
diff --git a/docker/constants.py b/docker/constants.py
index 71e543e530..8e7350d3d9 100644
--- a/docker/constants.py
+++ b/docker/constants.py
@@ -1,7 +1,7 @@
 import sys
 from .version import __version__
 
-DEFAULT_DOCKER_API_VERSION = '1.43'
+DEFAULT_DOCKER_API_VERSION = '1.44'
 MINIMUM_DOCKER_API_VERSION = '1.21'
 DEFAULT_TIMEOUT_SECONDS = 60
 STREAM_HEADER_SIZE_BYTES = 8
diff --git a/tests/Dockerfile-ssh-dind b/tests/Dockerfile-ssh-dind
index 2b7332b8b6..250c20f2c8 100644
--- a/tests/Dockerfile-ssh-dind
+++ b/tests/Dockerfile-ssh-dind
@@ -1,7 +1,7 @@
 # syntax=docker/dockerfile:1
 
-ARG API_VERSION=1.43
-ARG ENGINE_VERSION=24.0
+ARG API_VERSION=1.44
+ARG ENGINE_VERSION=25.0
 
 FROM docker:${ENGINE_VERSION}-dind
 

From dd82f9ae8e601d3c82ef8d4f2dbab0c16109152f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pawe=C5=82=20Gronowski?= <pawel.gronowski@docker.com>
Date: Thu, 7 Mar 2024 13:12:45 +0100
Subject: [PATCH 3/3] Bump minimum API version to 1.24
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

25.0 raised the minimum supported API verison: https://github.com/moby/moby/pull/46887

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>
---
 docker/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docker/constants.py b/docker/constants.py
index 8e7350d3d9..213ff61ed1 100644
--- a/docker/constants.py
+++ b/docker/constants.py
@@ -2,7 +2,7 @@
 from .version import __version__
 
 DEFAULT_DOCKER_API_VERSION = '1.44'
-MINIMUM_DOCKER_API_VERSION = '1.21'
+MINIMUM_DOCKER_API_VERSION = '1.24'
 DEFAULT_TIMEOUT_SECONDS = 60
 STREAM_HEADER_SIZE_BYTES = 8
 CONTAINER_LIMITS_KEYS = [
